versioned/storage/bigtable/src/main/java/org/projectnessie/versioned/storage/bigtable/BigTablePersist.java

```diff
@@ -568,40 +568,50 @@ public CloseableIterator<Obj> scanAllObjects(
   private class ScanAllObjectsIterator extends AbstractIterator<Obj>
       implements CloseableIterator<Obj> {
 
-    private final Predicate<ObjType> filter;
     private final Query.QueryPaginator paginator;
     private Iterator<Row> iter;
 
     private ByteString lastKey;
 
     ScanAllObjectsIterator(Predicate<ObjType> filter) {
-      this.filter = filter;
 
       Query q = Query.create(backend.tableObjs).prefix(keyPrefix);
 
       Filters.ChainFilter filterChain =
-          FILTERS.chain().filter(FILTERS.key().regex(keyPrefix.concat(REPO_REGEX_SUFFIX)));
-
-      // TODO the following filter does not really work as expected, can implement it later though.
-      //
-      // Filters.InterleaveFilter typeFilter = FILTERS.interleave();
-      // boolean all = true;
-      // for (ObjType type : ObjType.values()) {
-      //  boolean match = filter.test(type);
-      //  if (match) {
-      //    typeFilter.filter(FILTERS.value().exactMatch(OBJ_TYPE_VALUES[type.ordinal()]));
-      //  } else {
-      //    all = false;
-      //  }
-      // }
-      // if (!all) {
-      //  q.filter(
-      //      FILTERS
-      //          .chain()
-      //          .filter(FILTERS.family().exactMatch(FAMILY_OBJS))
-      //          .filter(FILTERS.qualifier().exactMatch(QUALIFIER_OBJ_TYPE))
-      //          .filter(typeFilter));
-      // }
+          FILTERS.chain().filter(FILTERS.family().exactMatch(FAMILY_OBJS));
+
+      Filters.InterleaveFilter typeFilter = null;
+      boolean all = true;
+      for (ObjType type : ObjType.values()) {
+        boolean match = filter.test(type);
+        if (match) {
+          if (typeFilter == null) {
+            typeFilter = FILTERS.interleave();
+          }
+          typeFilter.filter(
+              FILTERS
+                  .chain()
+                  .filter(FILTERS.qualifier().exactMatch(QUALIFIER_OBJ_TYPE))
+                  .filter(FILTERS.value().exactMatch(OBJ_TYPE_VALUES[type.ordinal()])));
+        } else {
+          all = false;
+        }
+      }
+      if (!all) {
+        // Edge case: the predicate did not match any object type
+        if (typeFilter == null) {
+          this.paginator = null;
```

Comment:
Isn't this rather an error than an edge case?

Reply:
It's up for debate. Do you prefer to throw here?

Reply:
It's either "all types" or an error IMO.

Reply:
OK for error then.

Reply:
It's rather an internal thingy - but an undefined situation ATM. Let's defer to another PR (I'd be in favor of falling back to "all types" actually).